src/librustc_typeck/diagnostics.rs

```diff
@@ -4567,6 +4567,52 @@ i_am_a_function();
 ```
 "##,
 
+E0619: r##"
+A not (yet) known type was used.
```

Comment:
I think we could say a bit *more* here. This message would perhaps be mildly easier to write if we knew *what* operation was giving it, but anyhow. (As an aside, this is (yet another...) of those messages that would be greatly improved by a "context sensitive" `--explain` mechanism, I think). 

```
The type-checker needed to know the type of an expression,
but that type had not yet been inferred. This is most often
fixed by adding a type annotation.
```

We could go on to give more context, but it may be overkill:

```
Type inference typically proceeds from the top of the
function to the bottom, figuring out types as it goes.
In some cases -- notably method calls and overloadable
operators like `*` -- the type checker may not have enough
information *yet* to make progress. This can be true even if the
rest of the function provides enough context (because the type-checker
hasn't looked that far ahead yet). In this case, type annotations can be
used to help it along. 
```